vllm/entrypoints/openai/serving_chat.py

```diff
@@ -225,7 +229,9 @@ async def create_chat_completion(
                 tool.model_dump() for tool in request.tools
             ]
 
-            prompt = self.tokenizer.apply_chat_template(
+            if self.chat_template is not None:
+                tokenizer.chat_template = self.chat_template
+            prompt = tokenizer.apply_chat_template(
```

Comment:
Perhaps we can pass `chat_template` to the `apply_chat_template` method instead of mutating the tokenizer?

Reply:
Good suggestion thanks, I missed that!